src/streaming/streamingclient.py

```diff
@@ -0,0 +1,178 @@
+
+        if not config:
+            raise ValueError('config must be provided')
+
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = \
+                'wss://api.rev.ai/speechtotext/{}/stream'.format(version)
+        self.on_error = on_error
+        self.on_close = on_close
+        self.on_open = on_open
+        self.thread = SpeechThread
+        self.client = websocket.WebSocket(enable_multithread = True, 
+            on_error = self.on_error, 
+            on_close = self.on_close, 
+            on_open = self.on_open
+            )
+
+    def start(self):
+        """Function to connect the websocket to the URL and start the response thread
+        """
+        url = self.base_url + '?access_token={}'.format(self.access_token) \
+              + '&content_type={}'.format(self.config.get_content_type_string()
+                )
+        try:
+            self.client.connect(url)
+            self.on_open()
+        except Exception as e:
+            self.client.abort()
+            self.on_error(e)
+
+    def end(self):
+        """Function to end the streaming service, close the websocket.
+        """
+        self.client.abort()
+
+    def send_data_as_gen(self, generator):
+        """Function to send binary audio data from a generator
+
+        :param generator: generator object that yields binary audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere 
+                    with the responses.""")
+
+        self.request_thread = self.thread(self.RequestSending, [generator])
+        self.request_thread.start()
+
+    def send_data_as_filepath(self, filepath):
+        """Function to send audio data from a given file
+
+        :param filepath: string containing file location
+        """
+        if not filepath:
+            raise ValueError('filepath must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere 
+                    with the responses.""")
+
+        self.send_data_as_gen(self.make_gen(filepath))
+
+    def make_gen(self, filepath):
+        """Get generator of audio data from a given filepath
+        
+        :param filepath: path to file including file name as string
+        :returns: generator of audio bytes
+        """
+        if not filepath:
+            raise ValueError('filepath must be provided')
+
+        with io.open(filepath, 'rb') as stream:
+            while True:
+                piece = stream.read(8192)
+                if not piece:
+                    break
+                yield piece
+
+    def RequestSending(self, generator):
```

Comment:
Formatting: The python convention for private methods is snake case starting with an underscore (ie `_request_sending`)

Reply:
I would also rename this method to be more clear. Something like `send_data`